### PR TITLE
external tax rates, close #1085

### LINKED ISSUE #1085: Externally Set Tax Rates

### PR DESCRIPTION
**Pull request by schleichardt**

### CODE REVIEW

File: commercetools-models/src/main/java/io/sphere/sdk/carts/CustomLineItemDraft.java [hunk 1]
```diff
@@ -27,19 +28,121 @@
 
     String getSlug();
 
+    /**
+     * Gets the tax rate for this custom line item.
```
Comment (gogregor) [inline commercetools-models/src/main/java/io/sphere/sdk/carts/CustomLineItemDraft.java]: The tax category will be used to get the tax rate for this custom line item when the cart tax mode is External.

File: commercetools-models/src/main/java/io/sphere/sdk/carts/LineItemDraft.java [hunk 2]
```diff
@@ -31,11 +32,20 @@
 
     Integer getVariantId();
 
+    /**
+     * Possible custom tax rate.
+     * If set, this tax rate will override the tax rate selected by the platform.
```
Comment (gogregor) [inline commercetools-models/src/main/java/io/sphere/sdk/carts/LineItemDraft.java]: Now the external tax rate does not override. External tax rates need to be set when the cart has the taxMode External.

File: commercetools-models/src/main/java/io/sphere/sdk/carts/commands/updateactions/SetCustomLineItemTaxRate.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
+package io.sphere.sdk.carts.commands.updateactions;
+
+import io.sphere.sdk.carts.Cart;
+import io.sphere.sdk.commands.UpdateActionImpl;
+import io.sphere.sdk.taxcategories.ExternalTaxRateDraft;
+
+import javax.annotation.Nullable;
+
+/**
+ Sets a new tax rate that overrides the tax rate selected by the platform. If `externalTaxRate` is empty, the platform will select a tax rate as in {@link AddLineItem}.
```
Comment (gogregor) [inline commercetools-models/src/main/java/io/sphere/sdk/carts/commands/updateactions/SetCustomLineItemTaxRate.java]: Also not correct anymore.

File: commercetools-models/src/main/java/io/sphere/sdk/carts/CustomLineItemDraft.java (hunk 1 above)
Comment (schleichardt) [inline commercetools-models/src/main/java/io/sphere/sdk/carts/CustomLineItemDraft.java]: you mean "Platform", right?

**Comment by schleichardt** [comment]
@lauraluiz please review

a Javadoc preview can be found here: http://schleichardt.github.io/external-tax-rate-preview/apidocs/io/sphere/sdk/carts/TaxMode.html

File: commercetools-models/src/main/java/io/sphere/sdk/carts/CustomLineItemDraft.java (hunk 1 above)
Comment (gogregor) [inline commercetools-models/src/main/java/io/sphere/sdk/carts/CustomLineItemDraft.java]: Right!

**Comment by lauraluiz** [comment]
👍

**Pull request merged**